File: src/main/java/com/github/unchama/seichiassist/listener/PlayerSeichiLevelUpListener.java
package com.github.unchama.seichiassist.listener;

import com.github.unchama.seichiassist.commands.*;
import com.github.unchama.seichiassist.data.*;
import com.github.unchama.seichiassist.event.*;
import com.github.unchama.seichiassist.util.*;
import org.bukkit.entity.*;
import org.bukkit.event.*;


/**
 * Created by karayuu on 2018/04/19
 */
public class PlayerSeichiLevelUpListener {
    @EventHandler
    public void onPlayerLevelUp(SeichiLevelUpEvent event) {
        Player p = event.getPlayer();

        switch (event.getLevelAfterLevelUp()) {
            case 10: Util.addItemToPlayerSafely(p, ItemData.getSuperPickaxe(5));
            case 20: {
                gachaCommand.Gachagive(p, 3, p.getName());
                gachaCommand.Gachagive(p, 10, p.getName());
            }
            case 30: Util.addItemToPlayerSafely(p, ItemData.getForLevelUpskull(p.getName(), 256));
            case 40: Util.addItemToPlayerSafely(p, ItemData.getGachaApple(256));
            case 50: gachaCommand.Gachagive(p, 27, p.getName());
            case 60: gachaCommand.Gachagive(p, 26, p.getName());
            case 70: gachaCommand.Gachagive(p, 25, p.getName());
            case 80: gachaCommand.Gachagive(p, 24, p.getName());
            case 90: gachaCommand.Gachagive(p, 20, p.getName());
            case 100: {
                gachaCommand.Gachagive(p, 21, p.getName());
                Util.addItemToPlayerSafely(p, ItemData.getElsa(1));
            }
        }
    }
}

File: src/main/scala/com/github/unchama/seichiassist/listener/PlayerSeichiLevelUpListener.scala
package com.github.unchama.seichiassist.listener

import cats.implicits._
import com.github.unchama.seichiassist.DefaultEffectEnvironment
import com.github.unchama.seichiassist.commands.legacy.GachaCommand
import com.github.unchama.seichiassist.data.{GachaSkullData, ItemData}
import com.github.unchama.seichiassist.event.SeichiLevelUpEvent
import com.github.unchama.seichiassist.util.Util.grantItemStacksEffect
import org.bukkit.entity.Player
import org.bukkit.event.{EventHandler, Listener}
import org.bukkit.inventory.ItemStack

object PlayerSeichiLevelUpListener extends Listener {
  @EventHandler
  def onPlayerLevelUp(event: SeichiLevelUpEvent): Unit = {
    implicit val player: Player = event.getPlayer

    val level = event.getLevelAfterLevelUp

    if (level != 30) giveItem("ガチャ券を付与する", level * 5, GachaSkullData.gachaForSeichiLevelUp)

    level match {
      case 10 =>
        giveItem("SuperPickaxeを配布する", 5, ItemData.getSuperPickaxe(1))
      case 20 =>
        GachaCommand.Gachagive(player, 3, player.getName)
        GachaCommand.Gachagive(player, 10, player.getName)
      case 30 =>
        giveItem("ガチャ券を付与する", level * 5, GachaSkullData.gachaForSeichiLevelUp)
      case 40 =>
        giveItem("ガチャりんごを付与する", 256, ItemData.getGachaApple(1))
      case 50 =>
        GachaCommand.Gachagive(player, 27, player.getName)
      case 60 =>
        GachaCommand.Gachagive(player, 26, player.getName)
      case 70 =>
        GachaCommand.Gachagive(player, 25, player.getName)
      case 80 =>
        GachaCommand.Gachagive(player, 24, player.getName)
      case 90 =>
        GachaCommand.Gachagive(player, 20, player.getName)
      case 100 =>
        GachaCommand.Gachagive(player, 21, player.getName)
        giveItem("エルサを付与する", 1, ItemData.getElsa(1))
    }
  }

  private def giveItem(context: String, amount: Int, item: ItemStack)(implicit player: Player): Unit = {
    DefaultEffectEnvironment.runEffectAsync(
      context,
      List.fill(amount)(
        grantItemStacksEffect(item).run(player)
      ).sequence
    )
  }
}
